watertap/unit_models/electrodialysis_0d.py

```diff
@@ -0,0 +1,938 @@
+            default={
+                "dynamic": False,
+                "has_holdup": False,
+                "property_package": self.config.property_package,
+                "property_package_args": self.config.property_package_args,
+            }
+        )
+        self.concentrate_channel.add_state_blocks(has_phase_equilibrium=False)
+        self.concentrate_channel.add_material_balances(
+            balance_type=self.config.material_balance_type, has_mass_transfer=True
+        )
+        self.concentrate_channel.add_momentum_balances(
+            balance_type=self.config.momentum_balance_type, has_pressure_change=False
+        )
+        # # TODO: Consider adding energy balances
+
+        # Add ports (creates inlets and outlets for each channel)
+        self.add_inlet_port(name="inlet_diluate", block=self.diluate_channel)
+        self.add_outlet_port(name="outlet_diluate", block=self.diluate_channel)
+        self.add_inlet_port(name="inlet_concentrate", block=self.concentrate_channel)
+        self.add_outlet_port(name="outlet_concentrate", block=self.concentrate_channel)
+
+        # Build Constraints
+        @self.Constraint(
+            self.flowsheet().config.time,
+            self.config.property_package.phase_list,
+            doc="Current-Voltage relationship",
+        )
+        def eq_current_voltage_relation(self, t, p):
+            surface_resistance_cp = (
+                self.membrane_surface_resistence["aem"]
+                + self.membrane_surface_resistence["cem"]
+                + self.spacer_thickness
+                / (
+                    0.5
+                    * (
+                        self.concentrate_channel.properties_in[
+                            t
+                        ].electrical_conductivity_phase[p]
+                        + self.concentrate_channel.properties_out[
+                            t
+                        ].electrical_conductivity_phase[p]
+                        + self.diluate_channel.properties_in[
+                            t
+                        ].electrical_conductivity_phase[p]
+                        + self.diluate_channel.properties_out[
+                            t
+                        ].electrical_conductivity_phase[p]
+                    )
+                )
+            )
+            return (
+                self.current[t]
+                * (
+                    surface_resistance_cp * self.cell_pair_num
+                    + self.electrodes_resistence
+                )
+                == self.voltage[t] * self.cell_width * self.cell_length
+            )
+
+        @self.Constraint(
+            self.flowsheet().config.time,
+            self.config.property_package.phase_list,
+            self.config.property_package.component_list,
+            doc="Equation for electrical migration flux_in",
+        )
+        def eq_elec_migration_flux_in(self, t, p, j):
+            if j == "H2O":
```

Comment:
I believe "H2O" is built as solvent always in the ion property pack. But I did add a `assert` for testing "H2O" in `component_list` in case other property packs are used.